gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+            flag to turn on/off the saving of the figure in a png file
+
+        Returns
+        -------
+        fig : `~matplotlib.figure.Figure`
+            figure with image of bin of the bg model for the
+            selected energy value
+        ax : `~matplotlib.axes.Axes`
+            axes of the figure
+        image : `~matplotlib.image.AxesImage`
+            image of the figure, optional
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import LogNorm
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+        energy = energy.flatten() # flatten
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy)
+        if nvalues != 1:
+            print("Expected exactly 1 value for energy, got {}.".format(nvalues))
+            raise IndexError
+        else:
+            energy = Quantity(energy[0])
+            print("Reqested plot only for 1 energy: {}".format(energy))
+
+        extent = self.image_extent
+        energy_bin_centers = self.spectrum_bin_centers
+
+        # find energy bin containing the specified energy
+        energy_bin, energy_bin_edges = self.find_energy_bin(energy)
+        ss_energy_bin_edges = "[{0}, {1}) {2}".format(energy_bin_edges[0].value,
+                                                      energy_bin_edges[1].value,
+                                                      energy_bin_edges.unit)
+        print("Found energy {0} in bin {1} with boundaries {2}.".format(energy,
+                                                                        energy_bin,
+                                                                        ss_energy_bin_edges))
+
+        # get data for the plot
+        ii = energy_bin
+        data = self.background[ii]
+        energy_bin_center = energy_bin_centers[ii]
+
+        # create plot
+        fig = plt.figure()
+        if ax is None:
+            ax = fig.add_subplot(111)
+
+        fig.set_size_inches(8., 8., forward=True)
+
+        image = ax.imshow(data.value,
+                          extent=extent.value,
+                          origin='lower', # do not invert image
+                          interpolation='nearest',
+                          norm=LogNorm(), # color log scale
+                          cmap='afmhot')
+
+        # set title and axis names
+        ax.set_title('Energy = [{0:.1f}, {1:.1f}) {2}'.format(energy_bin_edges[0].value,
+                                                              energy_bin_edges[1].value,
+                                                              energy_bin_edges.unit))
+        ax.set_xlabel('X / {}'.format(extent.unit))
+        ax.set_ylabel('Y / {}'.format(extent.unit))
+
+        # draw color bar
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes("right", size="5%", pad=0.05)
+        fig.colorbar(image, cax=cax, label='Bg rate / {}'.format(data.unit))
+
+        # save output
```

Comment:
Remove this `save` option!

It might be convenient for exactly your purpose.
But it's not flexible enough ... maybe I want to use PDF output or name the file something else.
It's better in library code to create flexible building blocks that enable others to do what they want.
In this case, just return the `ax` and `fig` object and the caller can save an extra line of code to file with a filename / extension / dpi resolution / ... of their choosing.